docs/old-reference-guide/modules/axon-framework-commands/pages/modeling/command-model.adoc

```diff
@@ -0,0 +1,236 @@
+= Command Model
+
+[NOTE]
+====
+In Axon Framework 5, the concept of "aggregates" has evolved into "command models" and "entities":
+
+* **Command Model** - The overall model that handles commands and maintains consistency
+* **Entity** - The technical implementation (Java class) of your command model
+
+This reflects the more flexible boundaries enabled by Dynamic Consistency Boundary (DCB).
+====
+
+The command model is an important concept in link:https://www.axoniq.io/concepts/domain-driven-design[Domain-Driven Design] and link:https://www.axoniq.io/concepts/cqrs-and-event-sourcing[CQRS].
+It represents the part of your system responsible for handling commands and maintaining business invariants.
+
+An **entity** is the technical implementation of your command model - a Java class containing command handlers and event sourcing handlers.
+Entities are clusters of related state and behavior acting as a single unit.
+They help maintain consistency and encapsulate domain logic within specific boundaries,
+ensuring changes to one part don't invalidate business rules.
+
+This chapter will cover the basics on how to implement a command model using entities in Axon Framework.
+
+== Entity identifiers
+
+Each entity has an identifier that uniquely identifies it within the context.
+Events published from this entity will contain this identifier,
+so that Axon Framework can later load the entity from its event stream.
+
+Commands that target entities contain a reference to this identifier.
+Axon Framework will load the events for the entity with this identifier,
+replay the events on an empty instance, and invoke the command. This is how the entity's state is reconstructed to support Event-Sourcing.
+
+[TIP]
+====
+The entity identifier has to be globally unique in your event store.
+This means that events are loaded based only on the identifier, and nothing else.
+The entity's type does not influence uniqueness at all.
+This means that if you have two entity types with the same identifier, Axon will
+load the events for both entities when handling a command.
+Mixing these events might lead to unexpected behavior, so we strongly recommend to design
+your entity identifiers with uniqueness in mind.
+====
+
+For creational commands that construct a new entity, the static `@CommandHandler` method returns the identifier.
+The identifier is then passed to the entity via the first event and stored in the `@EntityCreator` constructor.
+This approach allows the identifier field to be declared as `final`, ensuring immutability.
+
+[NOTE]
+.Routing commands in distributed environments
+====
+For command routing in distributed environments (for example, through Axon Server), use the `@RoutingKey` annotation on the command field that should be used for routing.
+
+If this annotation doesn't work for your use case, a different `RoutingStrategy` can be configured, as explained in the xref:infrastructure.adoc#routing-strategy[Routing Strategy] section.
+====
```

Comment:
the `@RoutingKey` annotation is deprecated as of 5.0, we should reference the `@CommandHandler#routingKey` attribute instead